pkg/version/base.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+// Base version information.
+//
+// This is the fallback data used when version information from git is not
+// provided via go ldflags (via Makefile).
+// TODO (irfanurrehman): Write a makefile for binaries which supplies these flags
+// As of now a static version value supplied in base.go is used instead.
+var (
+	// TODO: The current plan as per the todo above is to populate these
+	// fields using ldflags as per the comments against each.
+	// Worst case (for example, as is now), this file can have static info.
+	version = "v0.0.1-alpha.0" // output of "git describe"
+	// the prerequisite is that the branch should be
+	// tagged using the correct versioning strategy.
+
+	gitCommit    = "$Format:%H$" // sha1 from git, output of $(git rev-parse HEAD)
```

Comment:
Where is this intended to be interpolated? Perhaps supplying an `unknown` value is a sane default.

Reply:
Updated the default to `unknown`
Updated a Makefile, where this is calculated and set into the bin!